### PR TITLE
[test] Add rom_e2e_boot_policy_valid

### PR DESCRIPTION
**Pull request by alphan**
This PR adds `rom_e2e_boot_policy_valid` and removes `e2e_bootup_no_rom_ext_signature` and `e2e_bootup_bad_ext_signature` (along with the host binary) since these cases are already covered.

Signed-off-by: Alphan Ulusoy <alphan@google.com>

### CODE REVIEW

File: sw/device/silicon_creator/rom/e2e/BUILD [hunk 1]
```diff
@@ -36,14 +36,12 @@ MSG_PASS = "PASS!"
...
 
-SLOTS = [
-    "a",
-    "b",
-]
 
-SLOT_A_OFFSET = "0x0"
 
-SLOT_B_OFFSET = "0x80000"
+SLOTS = {
```
Comment (engdoreis) [inline sw/device/silicon_creator/rom/e2e/BUILD:53]: Nit. Maybe call it 'SLOTS_OFFSET?'

File: sw/device/silicon_creator/rom/e2e/BUILD [hunk 2]
```diff
@@ -53,8 +55,8 @@ SLOTS = {
     "b": "0x80000",
 }
 
-opentitan_flash_binary(
-    name = "empty_test_slot_a",
+[opentitan_flash_binary(
+    name = "empty_test_slot_{}".format(slot),
```
Comment (engdoreis) [inline sw/device/silicon_creator/rom/e2e/BUILD:59]: Nit. Just a matter of style.
```suggestion
    name = f"empty_test_slot_{slot}",
```

File: sw/device/silicon_creator/rom/e2e/BUILD [hunk 3]
```diff
@@ -687,3 +689,75 @@ test_suite(
         slot,
     ) for t in BOOT_POLICY_BAD_MANIFEST_CASES for slot in SLOTS],
 )
+
+[genrule(
+    name = "empty_test_slot_{}_corrupted_{}_bin_signed_{}".format(slot, device, key),
```
Comment (engdoreis) [inline sw/device/silicon_creator/rom/e2e/BUILD:662]: Nit.
I think it easier to read this way:
```suggestion
    name = f"empty_test_slot_{slot}_corrupted_{device}_bin_signed_{key}",
```

**Review by engdoreis — COMMENTED** [review]
Thanks @alphan for the PR. LGTM  but I'm not proeficient enough in Bazel to approve it. So I just left some minor comments,

File: sw/device/silicon_creator/rom/e2e/BUILD (hunk 2 above)
Comment (alphan) [reply sw/device/silicon_creator/rom/e2e/BUILD:59]: unfortunately bazel does not support f-strings.

File: sw/device/silicon_creator/rom/e2e/BUILD (hunk 3 above)
Comment (alphan) [reply sw/device/silicon_creator/rom/e2e/BUILD:662]: unfortunately bazel does not support f-strings.

**Comment by alphan** [comment]
Thanks @engdoreis!

**Comment by alphan** [comment]
@timothytrippel read for review

**Review by timothytrippel — APPROVED** [review]

**Pull request merged**